crates/bevy_render/src/lib.rs

```diff
@@ -236,3 +242,45 @@ fn check_for_render_resource_context(context: Option<Res<Box<dyn RenderResourceC
         );
     }
 }
+
+fn window_icon_changed(
+    mut map: Local<HashMap<WindowId, Handle<Texture>>>,
+    textures: Res<Assets<Texture>>,
+    mut windows: ResMut<Windows>,
+    asset_server: ResMut<AssetServer>,
+) {
+    for window in windows.iter_mut() {
+        /* Insert new icon changed */
+        for command in window.command_queue() {
```

Comment:
I'm not the biggest fan of this approach. 
We shouldn't be exposing the Window's internal `CommandQueue` to the outside world. 
And then iterating through the commands feels like a mixture of dependencies that should be separate.

Reply:
I dont like it either. This also leads to "empty code" in winit, because it can not do anything with this command if it is only a path and "moments" later it gets another event, but now with the bytes :/. 

The nice thing about this is that the command are automatically drained afterwards, so we do not have to keep track of whether the path just changed or not. (The downside is that we now have a timing dependency on the winit system that drains the commands).

Ideally I would just like an event, but in set_icon I have no access to world.